### DISCUSSION TITLE
Add ekg-auto-save

### DISCUSSION THREAD

**Discussion by qingshuizheng** [ahyatt/ekg#119] [Ideas]
Hello @ahyatt 

In #94, [garyo](https://github.com/garyo) asked if ekg has an auto-save mechanism like log-seq:

Q:
> How about auto-saving notes as I type, rather than C-c C-c which saves and closes?

A:
> For auto-saving notes, we don't have auto-save, but the develop branch does have drafts, so you can just save things manually as you go. Probably emacs has some horizontal autosave functionality that you could use for this, and it could be made to work with drafts.

So I wrote a auto-save function: https://github.com/ahyatt/ekg/pull/118

The idea (and the code) is mainly taken/adapted from `super-save`.

The code will auto-save ekg notes when:

1. emacs has been idle for x seconds
2. specific command is triggerred, e.g., `previous-buffer` `next-buffer`, etc
3. when specific hook is triggerred, e.g., `kill-buffer-hook`, so accidentally kill a buffer won't lose data, they are saved to drafts.

The code has user customizable options:

1. ekg-auto-save-when-idle
2. ekg-auto-save-idle
3. ekg-auto-save-silently
4. ekg-auto-save-triggers
5. ekg-auto-save-hook-triggers
6. ekg-auto-save-as-draft
7. ekg-auto-save-predicates

If you see it fit for this project, welcome to merge it or take any part of it.

Zheng.

**Comment by ahyatt** [comment]
This is great, thank you for contributing it!  I left a few comments in a code review, but I'm looking forward to this being in our next major release (which I should release in a week or two).
